v2/internal/test/ca.go

```diff
@@ -76,6 +79,41 @@ func (ca *CA) Bundle(td spiffeid.TrustDomain) *x509bundle.Bundle {
 	return bundle
 }
 
+func (ca *CA) PublicKey() crypto.PublicKey {
+	return ca.key.Public()
```

Comment:
How would you feel about generating a separate key for JWT signing purposes? Although it doesn't really matter for testing, we don't want people using the same key in practice and I'd hate for somebody to use this as reference code for building a CA.

Reply:
For a while I had one separated key, but I was unsure whether it was necessary or not for a fake. I will make the change :)